### ISSUE TITLE
Incorrect character positioning

### ISSUE THREAD

**Issue by jimryan** [yob/pdf-reader#261]
I'm working on a project that requires us to identify the position of some text within a PDF.  Occasionally we run into a problem where the positioning of text is off.  I've read through the relevant parts of the PDF spec, the instructions in the problematic PDFs, and the `pdf-reader` code, and can't seem to figure out where the problem lies.  `pdf-reader` seems to be doing the math right, yet the positioning of each character is off.

I've written some tools to help debug this and created a pared down version of one of our problem PDFs.  [Here is a screenshot](https://drive.google.com/open?id=16OT_Vq2M2IgcVJp2xlTCknJ3eakAuWrN) with each character that `pdf-reader` identifies and its position overlaid on an image of the PDF page.  You'll notice that the text towards the top is off.  Each character is either slightly too wide or too narrow, which makes the next character's starting position slightly off, eventually ending up with characters being way off.  Further down in the document ("Assorted Pork Chops" for example) is spot on, so when this happens, it isn't document- or even page-wide.

Here is the pared down PDF used in the above screenshot: https://drive.google.com/open?id=1SwCN5mjSU_mRI2wc-x8Noid8bn2epF8N

I have the original PDF if that could be useful.

Hopefully someone around here with more knowledge of PDFs/`pdf-reader` can point me in the right direction.  It looks like the width of each character is off, but I'm not sure why.

**Comment by yob** [comment]
This is a really great bug report - I love the image that demonstrates the issue.

I've looked into things a bit, but Unfortunately I'm at the edge of my knowledge on fonts and PDF matrixes, so I haven't made much progress on identifying the root cause yet. There's definitely some font size and positioning bugs in there as I've had a few similar sounding reports over the years.

**Comment by jimryan** [comment]
Thanks - I tried to compile everything that I had discovered up until reaching (much like you) the limits of my knowledge of fonts and PDF matrixes (I hadn't done matrix math since college, so that was a "fun" trip down memory lane).  

Please let me know how else I can be helpful in finding the cause of these issues.  If you have any ideas that you haven't had time to explore, feel free to pass them on to me.

**Comment by yob** [comment]
I went back and took another look at the sample PDF included in this issue.

Turns out, the gross glyph positioning errors (eg. where the glyphs are positioned off the right edge of the page) are fixed by #403, included ind the 2.7.0 release 🎆

I suspect there's still small errors in glyph positioning, things like the way we process kerning between glyphs might be off. However, given the main problem in this issue is fixed I'm going to close it. Sorry it took so many years to find the bug!
